test/framework/docker.go

```diff
@@ -29,7 +29,11 @@ func GetPublishedDockerPorts(
 		if len(addresses) < 1 {
 			return nil, errors.Errorf("there are no addresses for port %d", port)
 		}
-		_, pubPortStr, err := net.SplitHostPort(addresses[0])
+		addr := addresses[0]
+		if strings.HasPrefix(addr, ":::") {
+			addr = "[::]:" + addr[3:]
+		}
+		_, pubPortStr, err := net.SplitHostPort(addr)
```

Comment:
A little comment to explain this would be nice

Reply:
Just added and updated the code.
This format is seen on CircleCI runs ([an example run](https://app.circleci.com/pipelines/github/jijiechen/kuma/794/workflows/597fa41c-4fc9-4dc2-be08-cc150b2280cb/jobs/1223))